Team_Challenge_ToolBox

FUNCION: tipifica_variables

In [1]:
#Descargamos las herramientas necesarias
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import toolbox_ML as tb
from scipy.stats import f_oneway, ttest_ind

In [ ]:
import importlib
importlib.reload(tb)

import pandas as pd
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

tb.tipifica_variables(df, 10, 0.2)

In [23]:
#En primer lugar descargamos nuestros datos. Vamos a usar los datos del titanic
url = 'https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv'
df_titanic = pd.read_csv(url)

# Para comprobar que carga bien
df_titanic.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [24]:
# Ver información básica
df_titanic.info()

# Aplicamos la función de la ToolBox que describe nulos
resumen = tb.describe_df(df_titanic)
display(resumen)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
Tipo de Dato,int64,int64,int64,object,object,float64,int64,int64,object,float64,object,object
Nulos,0,0,0,0,0,177,0,0,0,0,687,2
% Nulos,0.0,0.0,0.0,0.0,0.0,19.86532,0.0,0.0,0.0,0.0,77.104377,0.224467
Valores Únicos,891,2,3,891,2,88,7,7,681,248,147,3


In [25]:
# Hacemos un copy
df = df_titanic.copy()

In [26]:
#Miramos porcentaje de Nulos
(df.isna().sum() / len(df) * 100).round(2).sort_values(ascending=False)

Cabin          77.10
Age            19.87
Embarked        0.22
PassengerId     0.00
Name            0.00
Pclass          0.00
Survived        0.00
Sex             0.00
Parch           0.00
SibSp           0.00
Fare            0.00
Ticket          0.00
dtype: float64

In [27]:
df["Cabin"].value_counts(dropna = False)

Cabin
NaN            687
G6               4
C23 C25 C27      4
B96 B98          4
F2               3
              ... 
E17              1
A24              1
C50              1
B42              1
C148             1
Name: count, Length: 148, dtype: int64

In [28]:
# Eliminamos Cabin ya que con su 77% de nulos imputar no seria lo correcto y no nos aporta nada 

df.drop(["Cabin"], axis = 1, inplace = True)

In [29]:
# Comprobamos que se eliminara correctamente
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(4)
memory usage: 76.7+ KB


- Seguimos con Embarked y analizamos que hacer con sus nulos:

In [31]:
#Como tiene 0.22% nulos le aplicamos la moda
embarked_mode = df["Embarked"].mode()[0]

In [33]:
df.loc[df["Embarked"].isna(), "Embarked"] = embarked_mode

- Continuamos con Age y vemos que con su 19.87% en nulos podemos:

In [35]:
# Ahora hacemos el analisis de "age" que es de tipo numerico
df["Age"].describe()

count    714.000000
mean      29.699118
std       14.526497
min        0.420000
25%       20.125000
50%       28.000000
75%       38.000000
max       80.000000
Name: Age, dtype: float64